converters/fromZigbee.js

```diff
@@ -531,7 +532,14 @@ const converters = {
                 result.color_options = {execute_if_off: ((msg.data.options & 1<<0) > 0)};
             }
 
-            return result;
+            // handle color property sync
+            // NOTE: this should the last thing we do so we would have processed all values
+            //       the assign here is important as we do not want to wipe out other attributes
+            if (colorSync) {
+                return Object.assign(result, syncColor(result, meta.state));
```

Comment:
I would also move the `Object.assign` into the `syncColor` method. So here the if/else will be replaced by just `return syncColor(result, meta.state);`

Reply:
Not sure we can move this, as we only want to do this in the fromZigbee case. In the toZigbee we just return an empty state when not doing a color_sync as we just want the data from the read/reports. Inside syncColor() we have no idea where we’re called from.

Reply:
I actually have an idea how to do this, let’s see if it actually works.

Reply:
> I actually have an idea how to do this, let’s see if it actually works.
It does not :(

Reply:
I ended up with `return Object.assign(result, syncColor(result, meta.state, options));` in the fromZigbee converter.

It still improved readability. Unless we pass another var indicating if we calling from toZigbee or fromZigbee, but that becomes less readable IMHO.

Some more background why we can't just do the assign inside syncColor(). When color_sync is false for the fromZigbee path we should return an unmodified payload so all reported/read attribute get published untouched. When color_sync is true, we pass color_temp, and color modified based on color_mode and all others unmodified. In theory we could just return newState inside syncColor(), but for the toZigbee path we don't want to return {} in the color_sync == false path so we avoid the need for a debounce to be configured. Because otherwise we get 2 publish... or 1 + 2*group members for a group. One with the set value and nearly instantly one with the value reported by the bulb(s).